src/Routes.php

```diff
@@ -56,7 +56,7 @@ public function setFormation($formation)
 
     public function setTypes(array $types = [])
     {
-        $this->types = $types;
+        $this->types['only'] = $types;
```

Comment:
Good catch, we should remove setTypes and adjust the route macro in the `FormationServiceProvider`

before
```php
Route::macro('formation', function ($resource, $formation, array $types = []) {
    return app(Routes::class)
        ->setResource($resource)
        ->setFormation($formation)
        ->setTypes($types);
});
```
after
```php
Route::macro('formation', function ($resource, $formation) {
    return app(Routes::class)
        ->setResource($resource)
        ->setFormation($formation)
});
```

Because it returns the Routes object.. they can chain only and except and not add the array param

Reply:
Okay, Just for confirmation,

Then this syntax will not be valid anymore. Right?

`Route::formation('posts.tags', TagFormation::class, ['sync'])->pivot();`

Uses will need to use always,

`Route::formation('posts.tags', TagFormation::class)->pivot()->only(['sync]);`

Reply:
@mitulgolakiya yup, confirmed. This is the intended method of limiting generated routes now

Reply:
Okay.